src/gtk/dirdlg.cpp

```diff
@@ -132,13 +143,24 @@ bool wxDirDialog::Create(wxWindow* parent,
 
 void wxDirDialog::GTKOnAccept()
 {
-    wxGtkString str(gtk_file_chooser_get_filename(GTK_FILE_CHOOSER(m_widget)));
-    m_selectedDirectory = wxString::FromUTF8(str);
+    GSList *fnamesi = gtk_file_chooser_get_filenames(GTK_FILE_CHOOSER(m_widget));
+    GSList *fnames = fnamesi;
+
+    while ( fnamesi )
+    {
+        wxString dir(wxString::FromUTF8(static_cast<gchar *>(fnamesi->data)));
+        m_paths.Add(dir);
+
+        g_free(fnamesi->data);
+        fnamesi = fnamesi->next;
+    }
+
+    g_slist_free(fnames);
 
     // change to the directory where the user went if asked
-    if (HasFlag(wxDD_CHANGE_DIR))
+    if ( !HasFlag(wxDD_MULTIPLE) && HasFlag(wxDD_CHANGE_DIR) )
     {
-        wxSetWorkingDirectory(m_selectedDirectory);
+        wxSetWorkingDirectory(m_paths.Last());
```

Comment:
Just checking: are we really guaranteed to have a path here? I.e. it's impossible to accept the dialog without choosing anything?

Reply:
When I test this on the dialogs sample on GTK3, it is giving the directory that the dialog is in as the returned directory if the user didn't select anything.